example/src/main/java/io/netty/example/http/router/BadClientSilencer.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.example.http.router;
+
+import io.netty.channel.ChannelHandler.Sharable;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+/**
+ * This utility handler should be put at the last position of the inbound pipeline to
+ * catch all exceptions caused by bad client (closed connection, malformed request etc.)
+ * and server processing.
+ *
+ * By default exceptions are logged to the internal logger. You may need to override
+ * onUnknownMessage, onBadClient, and onBadServer to log to more suitable places.
+ */
+@Sharable
+public class BadClientSilencer extends SimpleChannelInboundHandler<Object> {
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(BadClientSilencer.class);
+
+    /** Logs to stderr. Override this method to log to other places if you want. */
```

Comment:
The javadoc is incorrect for all the three methods as it not log to stderr but to the logger